### ISSUE TITLE
Question: Is it a best practice to periodically reset summaries on a timer?

### ISSUE THREAD

**Issue by swilliams-a3digital** [siimon/prom-client#195]
Hi, we just started using this library and prometheus.  It is very easy to use... thanks!  We are creating a summary of request response times and have a concern that if we keep collecting data for the summary, after days / weeks of adding observations it will consume a lot of memory.  Our assumption is you have to hang onto each observation in order to do the percentiles and the millions of observations will eat up memory. Our thought is we should run a timer and periodically reset the summary.  Is this a valid concern?  Is this wise?  Any advice from people doing something similar is greatly appreciated.

Thanks
Scott

**Comment by siimon** [comment]
I don't think you need to worry about that. We are using tdigest for percentiles calculations and it's a pretty smart algorithm, reading that paper is recommended!

In general resetting should not be used in production, the only use case I see for that is in testing.
